http/restful-ws/src/main/java/io/cloudevents/http/restful/ws/CloudEventsProvider.java

```diff
@@ -0,0 +1,149 @@
+/*
+ * Copyright 2018-Present The CloudEvents Authors
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package io.cloudevents.http.restful.ws;
+
+import io.cloudevents.CloudEvent;
+import io.cloudevents.format.EventFormat;
+import io.cloudevents.format.EventFormatProvider;
+import io.cloudevents.http.restful.ws.impl.RestfulWSClientMessageVisitor;
+import io.cloudevents.http.restful.ws.impl.RestfulWSMessageFactory;
+import io.cloudevents.http.restful.ws.impl.RestfulWSMessageVisitor;
+import io.cloudevents.http.restful.ws.impl.Utils;
+import io.cloudevents.message.BinaryMessageVisitor;
+import io.cloudevents.message.MessageVisitor;
+
+import javax.ws.rs.Consumes;
+import javax.ws.rs.Produces;
+import javax.ws.rs.WebApplicationException;
+import javax.ws.rs.client.ClientRequestContext;
+import javax.ws.rs.client.ClientRequestFilter;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.MultivaluedMap;
+import javax.ws.rs.ext.MessageBodyReader;
+import javax.ws.rs.ext.MessageBodyWriter;
+import javax.ws.rs.ext.Provider;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.lang.annotation.Annotation;
+import java.lang.reflect.Type;
+import java.util.Arrays;
+import java.util.Optional;
+
+@Provider
+@Consumes(MediaType.WILDCARD)
+@Produces(MediaType.WILDCARD)
+public class CloudEventsProvider implements MessageBodyReader<CloudEvent>, MessageBodyWriter<CloudEvent>, ClientRequestFilter {
```

Comment:
Note that in Quarkus you'd have to implement `AsyncMessageBodyWriter` if you ever plan on supporting sending cloud events from the IO thread (for async requests).

Reply:
That's a RESTEasy specific API.

Reply:
I think this kind of integration should not live in this project, but maybe in another submodule